test/02_class/class.h

```diff
@@ -51,5 +51,16 @@ class DefaultArguments
   int add(int i = 1, int j = 2) const;
 };
 
+class NonPublicParamInConstructor
+{
+protected:
+  struct ProtectedData{};
+
+public:
+  NonPublicParamInConstructor(const ProtectedData&) {}
+  NonPublicParamInConstructor(const std::string& name) : m_name(name) {}
+  std::string m_name;
```

Comment:
Let's add a call in the `.py` unit test to make sure we can instantiate this class via the public constructor, just in case our check breaks and it removes _all_ constructors.

Reply:
Done